src/dune_rules/coq/coq_rules.ml

```diff
@@ -718,13 +721,13 @@ let coq_modules_of_theory ~sctx lib =
     Coq_sources.library coq_sources ~name |> List.rev_map ~f:Coq_module.source
 ;;
 
-let source_rule ~sctx theories =
+let source_rule ~sctx extra_sources theories =
```

Comment:
I think we should also include the `extra_sources` of the theories we depend on, maybe time to introduce a `coq_sources_of_theory` function?